astrix-context/src/main/java/com/avanza/astrix/beans/service/ServiceBeanInstance.java

```diff
@@ -118,6 +118,10 @@ public void renewLease() {
 				return;
 			}
 			if (serviceHasChanged(serviceDiscoveryResult.getResult())) {
+				if (isBound() && currentProperties != null) {
+					log.info("Service properties for bean={} astrixBeanId={} have changed, will rebind service bean.", getBeanKey(), id);
+					destroy();
```

Comment:
If I understand this correctly, calling `destroy` here will cause the `ServiceBeanInstance` to be rebound. So it is not really destroy in the sense that it can no longer be used, but rather unbound. If this is the case, I suggest we rename the method and rephrase the logging a bit to reflect this.

Reply:
Have added another commit now that makes such changes.